### PR TITLE
feat(question): create draft

### LINKED ISSUE #2966: [QuestionCreate] Publish or save as Draft
It should be possible to save an item as either a Draft or published document.

### PR DESCRIPTION
**Pull request by onim-at**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

- creates and modify a draft question. in case of draft validates only the question title
- added draft icon to question list and question page
- fix: on question update keeps the original _createdBy user

## Git Issues

Closes #2966 

## Screenshots/Videos

image
image
image

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of our monthly maintainers call (first Monday of the month)

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

File: src/pages/Question/QuestionPage.tsx [hunk 1]
```diff
@@ -57,6 +57,11 @@ export const QuestionPage = (props: IProps) => {
               onUsefulClick={onUsefulClick}
             />
           </Flex>
+          <ModerationStatus
+            status={question.moderation}
+            contentType="howto"
```
Comment (evakill) [inline src/pages/Question/QuestionPage.tsx:62]:
```suggestion
            contentType="question"
```
copy pasta, as i call it 🍝

File: src/pages/Question/question.routes.test.tsx [hunk 2]
```diff
@@ -185,6 +185,47 @@ describe('question.routes', () => {
...
         description: 'Question description',
         tags: {},
+        allowDraftSave: false,
+        moderation: 'accepted',
+      })
+
+      expect(history.location.pathname).toBe('/question/question-title')
+    })
+
+    it('allows user to draft a question', async () => {
```
Comment (evakill) [inline src/pages/Question/question.routes.test.tsx:195]: thanks for the tests 👏

**Review by evakill — COMMENTED** [review]
looks great!! thanks @onim-at - one small change then this is good to go

File: src/pages/Question/QuestionPage.tsx (hunk 1 above)
Comment (onim-at) [reply src/pages/Question/QuestionPage.tsx]: whoopsie 🍝

**Comment by onim-at** [comment]
@evakill not sure why the test failed, it passed locally and restarting the pipeline seems to have fixed it

**Review by evakill — APPROVED** [review]

**Pull request merged**